### ISSUE TITLE
googlephotos: 404 on "odd" file

### ISSUE THREAD

**Issue by philips** [rclone/rclone#4144]
#### What is the problem you are having with rclone?

Getting a 404 on an "odd" source object name.

```
{"level":"error","msg":"Failed to copy: failed to open source object: \ufffdPNG\r\n\u001a\n\u0000\u0000\u0000\rIHDR\u0000\u0000\u0000\ufffd\u0000\u0000\u0000\ufffd\u0008\u0003\u0000\u0000\u0000\ufffd\ufffd^\ufffd\u0000\u0000\u0000\u0003sBIT\u0008\u0008\u0008\ufffd\ufffdO\ufffd\u0000\u0000\u0000*PLTEGpL___\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd]]]___aaa\ufffd\ufffd\ufffdttt\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffdw\ufffd\u0000\u0000\u0000\rtRNS\u0000f}\ufffdڋ\ufffdR1\u0015\ufffdo\ufffd\ufffd*MN\u0000\u0000\u0002\ufffdIDATx\ufffd\ufffd\ufffd\ufffdr\ufffd0\u0010@\ufffd.A#\u0001\ufffd\ufffdwk\u0002\n\ufffd\u0004\ufffd\ufffd\u000c\ufffd\ufffd{f\ufffd\ufffd\u000b\ufffdm\u0012\ufffd\ufffd\ufffd\u0007\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\ufffd\ufffd⻳\ufffd\ufffd\u001b\ufffdܝ}\u0011_\ufffd\ufffd9ğ}\u0019\ufffd\u0013\ufffd\ufffd\ufffd\u0011\ufffd\ufffd\\7\t\ufffd8\ufffd\ufffd|\u0012ߌQ\ufffd\ufffd\ufffdW򙛴i\ufffd\ufffd1\ufffd\ufffd_\ufffdC\ufffd\ufffd\ufffdc;I\n\ufffd\ufffdX\ufffdb:I\n\u0012\ufffd\ufffd\ufffdd\u000e\u0012w`\ufffdI\ufffd b=\ufffd-\ufffd\ufffdA\ufffd'IgHH\ufffd(\ufffd\ufffd\ufffd\ufffd\ufffdI\ufffd\ufffd;\ufffd \ufffdi\u000e\ufffdI6A,'\ufffd\u00041\ufffd$\u000bb7I\u0016\ufffdl\ufffd]\u0010\ufffdIvAD.\u0016\ufffdLA.\ufffdAL\u0026\ufffd}\u000c\ufffdo\u00071\ufffdJ\ufffd'\ufffd\u003c\ufffd\ufffd$\ufffdOo~\ufffd\ufffd*\ufffd\u0004\ufffd\ufffd\ufffd\u0012\ufffd\ufffd*\ufffd\u0006\ufffd\ufffdd\u0015d\\1\ufffdJ\ufffdA\ufffdAzKI\ufffd }q\u0010KI\ufffd ô\u0026\ufffde\ufffd\ufffd\ufffd*Y\u0007)\u000cbf\ufffd\ufffddH\ufffdym\u00103I\ufffd\ufffdϐ\ufffd\u000c\u0012l\ufffd%)\ufffdc\ufffd\u0014\u0007\ufffd\ufffdh\ufffd3\u0005\ufffd\ufffd\u00061\ufffdqM+\ufffd\u0019D.\ufffd \ufffdIo!I\u0016\ufffd\u003cHLr՝$\u000fR\u001e$\ufffde\ufffd\ufffd\ufffd\ufffd n\ufffd\ufffd0;؍$\ufffd\u0005\ufffd\u000c\ufffdV\ufffd\ufffd$\ufffd \ufffdA\ufffd')\u0004Y\ufffd\ufffd\ufffd\ufffd,\ufffdI|z\ufffdֿ3\ufffd\ufffd$]ܣ\\\ufffdw\u0006Q\ufffd\ufffd?\u001e}\ufffd=\ufffd\ufffdl@\ufffdIR\ufffd\ufffdz\ufffd\ufffd(\u003eKJ+D\ufffd\ufffdO\ufffd\ufffd\u0005\ufffdI\ufffdA\ufffd\ufffd}\ufffd\ufffd\ufffd\ufffd$\ufffd\u0015r4\ufffd\ufffd$\ufffd Ҵ\ufffd݈J7\ufffdJ\ufffd#*\ufffdLA\ufffd\ufffd\ufffd^{\ufffd\ufffd \ufffd\u0010ofZ\ufffdt\ufffd\ufffd6\u001b\\\\\ufffd.\ufffd}ӈ\ufffd6.\ufffd\ufffd\ufffdx׫\ufffd\ufffd3\ufffd\ufffd\ufffd\ufffd9\u0008R\ufffd\ufffdZ^Քė\u001e}NʟG\ufffd\u00269\u0008\ufffdb\u0010eI\ufffd+D\ufffd=\ufffdpŗU%9\nR\ufffd\ufffdU\ufffd\ufffd(ȴŎ\ufffdﺴ\u00269\u000cr\ufffd$\ufffd/$Be\u000eMgI\ufffd\ufffd5ާ\u0026ɋ \ufffdiIr\ufffd,\ufffdT\ufffd.\ufffd\ufffdm\ufffd\u0001\ufffd\u000e\ufffd\ufffd\ufffdܶOn\ufffd\ufffd\ufffd^\ufffdՙ\ufffdw\ufffdF\ufffd\ufffdA\ufffd\ufffd'I\ufffd\ufffd\ufffd\ufffd|\ufffdٿZ\ufffd\ufffd\ufffd\u001c\ufffd\u003e'\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\ufffd#\ufffd¤\u003c\u0019U\ufffda\ufffd\u0000\u0000\u0000\u0000IEND\ufffdB`\ufffd (404 404 Not Found)","object":"2012/2012-11/Scan 1.tiff","objectType":"*googlephotos.Object","source":"operations/operations.go:451","time":"2020-04-17T07:49:13.752678+00:00"}
```

#### What is your rclone version (output from `rclone version`)

1.51.0

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux ARM

####  Which cloud storage system are you using? (eg Google Drive)

Google Photos

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

copy photos:/media/by-month/ /data

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

I am generating this now but I need to figure out how to filter it down to things that are relevant.

**Comment by philips** [comment]
Found the exact file causing the issue. can reproduce easily now.

```
rclone copy -vv photos:/media/by-month/2012/2012-11/Scan\ 1.tiff /tmp/foo.tiff
```

```
2020/04/17 20:53:13 DEBUG : Using config file from "/config/rclone/rclone.conf"
2020/04/17 20:53:13 DEBUG : Google Photos path "media/by-month/2012/2012-11": NewObject: remote="Scan 1.tiff"
2020/04/17 20:53:13 DEBUG : Google Photos path "media/by-month/2012/2012-11": List: dir=""
2020/04/17 20:53:14 DEBUG : Google Photos path "media/by-month/2012/2012-11": >List: err=<nil>
2020/04/17 20:53:14 DEBUG : Google Photos path "media/by-month/2012/2012-11": >NewObject:
2020/04/17 20:53:14 DEBUG : Google Photos path "media/by-month/2012/2012-11": NewObject: remote="Scan 1.tiff"
2020/04/17 20:53:14 DEBUG : Google Photos path "media/by-month/2012/2012-11": List: dir=""
2020/04/17 20:53:15 DEBUG : Google Photos path "media/by-month/2012/2012-11": >List: err=<nil>
2020/04/17 20:53:15 DEBUG : Google Photos path "media/by-month/2012/2012-11": >NewObject:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: Need to transfer - File not found at Destination
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: Open:
2020/04/17 20:53:15 DEBUG : Scan 1.tiff: >Open:
2020/04/17 20:53:15 ERROR : Scan 1.tiff: Failed to copy: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:15 ERROR : Attempt 1/3 failed with 1 errors and: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:15 DEBUG : Google Photos path "media/by-month/2012/2012-11": NewObject: remote="Scan 1.tiff"
2020/04/17 20:53:15 DEBUG : Google Photos path "media/by-month/2012/2012-11": List: dir=""
2020/04/17 20:53:16 DEBUG : Google Photos path "media/by-month/2012/2012-11": >List: err=<nil>
2020/04/17 20:53:16 DEBUG : Google Photos path "media/by-month/2012/2012-11": >NewObject:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: Need to transfer - File not found at Destination
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: Open:
2020/04/17 20:53:16 DEBUG : Scan 1.tiff: >Open:
2020/04/17 20:53:16 ERROR : Scan 1.tiff: Failed to copy: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:16 ERROR : Attempt 2/3 failed with 1 errors and: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:16 DEBUG : Google Photos path "media/by-month/2012/2012-11": NewObject: remote="Scan 1.tiff"
2020/04/17 20:53:16 DEBUG : Google Photos path "media/by-month/2012/2012-11": List: dir=""
2020/04/17 20:53:17 DEBUG : Google Photos path "media/by-month/2012/2012-11": >List: err=<nil>
2020/04/17 20:53:17 DEBUG : Google Photos path "media/by-month/2012/2012-11": >NewObject:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: Need to transfer - File not found at Destination
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: Size:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: >Size:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: Open:
2020/04/17 20:53:17 DEBUG : Scan 1.tiff: >Open:
2020/04/17 20:53:17 ERROR : Scan 1.tiff: Failed to copy: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:17 ERROR : Attempt 3/3 failed with 1 errors and: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
2020/04/17 20:53:17 Failed to copy: failed to open source object: �PNG

tRNSf}�ڋ�R1�o��*MN�IDATx����r�0@�.A#��wk]]]___aaa���ttt��������w�
���
   ��{f��
         �m�����⻳��_��9ğ}������\7      �8��|ߌQ���W򙛴i��1��_�C���c;I
��X�b:I
���dw`�I� b=�-��A�'IgHH�(�����I��;� �i�I6A,'�1�$
                                                b7I�l�]�IvAD.�LA.�AL&�}
                                                                       �o1�J�'�<��$�Oo~��*������*���dd\1�J�A�AzKI� }qKI� ô&�e���*Y)
          bf��dH�ym3I��ϐ�
                         l�%)�c���h�3��1�qM+�D.� �Io!I�<HLr՝$R$�e���� n��0;؍$��
                                                                               �V��$� �A�')Y����,�I|z�ֿ3��$]ܣ\�wQ��?}�=��l@�IR��z��(>KJ+D��O���I�A��}����$�r4��$� Ҵ�݈J7�J�#*�LA���^{�� �ofZ�t��6\�.�}ӈ�6.���x׫��3����R��Z^Քė}NʟG�&�beI�+D�=�pŗU%9
R��U��(ȴŎ�ﺴ&9
             r�$�/$BeMgI��5ާ&ɋ �iIr�,�T�.��m�����ܶOn���^�ՙ�w�F��A��'I����|�ٿZ����>'�#�¤<U�a�IEND�B`� (404 404 Not Found)
status code: 1

```

**Comment by philips** [comment]
something else that is interesting. I downloaded this image and uploaded it to a different account. Google Photos renders it as a grey error box and it doesn't show up in rclone ls.

![Screenshot 2020-04-17 at 2 24 23 PM](https://user-images.githubusercontent.com/33544/79615341-6334da00-80b7-11ea-93e5-4cbcec3afa8a.png)

**Comment by ncw** [comment]
> Found the exact file causing the issue. can reproduce easily now.

That is good news!

If nothing else we need to fix the error message!

Can you run your command with `-vv --dump bodies --log-file rclone.log` and attach rclone.log to this issue please?

That will show us exactly what is going on - there is some error path which is very kooky and that might give a clue as how to fix.

**Comment by philips** [comment]
[rclone.log](https://github.com/rclone/rclone/files/4527262/rclone.log)

See attached log. Note I redacted a bunch of media items in the list manually that are unrelated to this issue.

**Comment by ncw** [comment]
Now that is weird!

We send this request

```
2020/04/24 07:02:47 DEBUG : HTTP REQUEST (req 0x25ba200)
2020/04/24 07:02:47 DEBUG : GET /lr/AFBm1_ZZgR0bL--_SfrM1caD7<snip>UX1LN2PqJh=d HTTP/1.1
Host: lh3.googleusercontent.com
User-Agent: rclone/v1.51.0
Authorization: XXXX
Accept-Encoding: gzip
```

And we get back this which is a 404 message with a Content-Type of `image/png`

```
2020/04/24 07:02:47 DEBUG : HTTP RESPONSE (req 0x25ba200)
2020/04/24 07:02:47 DEBUG : HTTP/2.0 404 Not Found
Content-Length: 832
Access-Control-Allow-Origin: *
Alt-Svc: quic=":443"; ma=2592000; v="46,43",h3-Q050=":443"; ma=2592000,h3-Q049=":443"; ma=2592000,h3-Q048=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,h3-T050=":443"; ma=2592000
Content-Type: image/png
Date: Fri, 24 Apr 2020 07:02:47 GMT
Server: fife
Timing-Allow-Origin: *
X-Content-Type-Options: nosniff
X-Xss-Protection: 0

�PNG
```

I decoded the png - here it is - look familiar? ;-)

![Error image](https://user-images.githubusercontent.com/536803/80236046-7c5aff00-8652-11ea-9205-ea6e53888b19.png)

I guess there is something up with that file and google can't find it so they return a placeholder image as the 404 message which makes sense.

Did you manage to download the file from google photos web interface? Or is it gone?

So I think what I need to do is get rclone to pay attention to the `Content-Type` of the 404 message and return a better textual error

Can you give this a go?

https://beta.rclone.org/branch/v1.51.0-231-g4b56beba-fix-4144-gphotos-failed-beta/ (uploaded in 15-30 mins)

**Comment by philips** [comment]
I will test it.

Indeed i have the original image. My hunch? They allowed tiffs originally but then decided to block them. I uploaded this image before this change so on download it blocks with a 404 and denies the upload too now.

**Comment by philips** [comment]
@ncw still getting the PNG binary straight into the log line:

```
2020/04/25 00:25:42 Failed to copy: failed to open source object: <89>PNG^M
... etc etc
```

**Comment by philips** [comment]
Also, I am happy to build myself if given a patch and I can play around
with what you were trying to do.

**Comment by ncw** [comment]
> still getting the PNG binary straight into the log line

:-(

> Also, I am happy to build myself if given a patch and I can play around
with what you were trying to do.

You'll find the patch on this branch: https://github.com/rclone/rclone/tree/fix-4144-gphotos-failed

Specifically here:

https://github.com/rclone/rclone/commit/4b56bebab84ac1cb2b759f677e03254b967aad2f

If you can figure out what I did wrong that would be great!

**Comment by philips** [comment]
@ncw oopsie! you used `ContentType` instead of `Content-Type`. Put up a PR with the fixed patch I tested here: https://github.com/rclone/rclone/pull/4174

**Comment by philips** [comment]
Tested output:

```
2020/04/27 19:41:47 Failed to copy: failed to open source object: Image not found or broken (404 404 Not Found)                   
```

**Comment by ncw** [comment]
Nice one thank you! I'll merge the PR now :-)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52

**Comment by philips** [comment]
perfect, thank you!
